src/client/components/dimension-tile/dimension-tile.tsx

```diff
@@ -330,7 +336,7 @@ export class DimensionTile extends React.Component<DimensionTileProps, Dimension
   getGranularityActions() {
     const { dimension } = this.props;
     const { selectedGranularity } = this.state;
-    var granularities = dimension.granularities || DEFAULT_DURATION_GRANULARITIES;
+    var granularities = dimension.granularities || getGranularities(dimension.kind as DimensionKind, dimension.bucketedBy, true);
```

Comment:
dimension.kind should already be type DimensionKind, could you update the type at the source